File: UTESA/DCE 5 Estrellas/dce2005/Estrella1/DCE1_Ejemplos/src/CS/Cliente/OpcionConfirmarVenta.cs
using System;
using System.Collections.Generic;
using System.Text;

using DCE05.Ejemplos.EstrellaUno.ReglasNegocio;

namespace DCE05.Ejemplos.EstrellaUno.Cliente {
    
    /// <summary>
    /// Representa la opción de finalizar una venta.
    /// </summary>
    internal class OpcionConfirmarVenta : Opcion {

        /// <summary>
        /// Ejecuta la acción asociada a la opción.
        /// </summary>
        /// <exception cref="OpcionInvalidaException">Si la opción no fue ejecutada exitosamente.</exception>
        internal override void EjecutarAccion() {
            if (PuntoDeVenta.VentaActual == null) {
                throw new OpcionInvalidaException("La venta no fue iniciada.");
            }

            try {
                CatalogoVentas catalogo = new CatalogoVentas();
                catalogo.ConfirmarVenta(PuntoDeVenta.VentaActual);
                Console.WriteLine("Venta confirmada.\nTotal: {0}\n", PuntoDeVenta.VentaActual.Total());
                PuntoDeVenta.VentaActual = null;
            } catch (ReglasNegocioException ex) {
                Console.WriteLine("Error al finalizar la venta: " + ex.Message);
            }
        }

        /// <summary>
        /// Construye una instancia de la opción con sus datos básicos.
        /// </summary>
        /// <param name="codigo">El código de la opción.</param>
        internal OpcionConfirmarVenta(int codigo) {
            Codigo = codigo;
            Descripcion = "Confirmar Venta Actual";
        }

    }
}

File: UTESA/DCE 5 Estrellas/dce2005/Estrella1/DCE1_Ejemplos/src/VB/Cliente/OpcionConfirmarVenta.vb
Imports DCE05.Ejemplos.EstrellaUno.ReglasNegocio

''' <summary>
''' Representa la opción de finalizar una venta.
''' </summary>
Public Class OpcionConfirmarVenta
    Inherits Opcion

    ''' <summary>
    ''' Ejecuta la acción asociada a la opción.
    ''' </summary>
    Friend Overrides Sub EjecutarAccion()
        If PuntoDeVenta.VentaActual Is Nothing Then
            Throw New OpcionInvalidaException("La venta no fue iniciada.")
        End If

        Try
            Dim Catalogo As CatalogoVentas = New CatalogoVentas()
            Catalogo.ConfirmarVenta(PuntoDeVenta.VentaActual)
            Console.WriteLine("Venta confirmada." + Chr(13) + " Total: {0}" + Chr(13), PuntoDeVenta.VentaActual.Total())
            PuntoDeVenta.VentaActual = Nothing
        Catch Ex As ReglasNegocioException
            Console.WriteLine("Error al finalizar la venta: " + Ex.Message)
        End Try
    End Sub

    ''' <summary>
    ''' Construye una instancia de la opción con sus datos básicos.
    ''' </summary>
    ''' <param name="Codigo">El código de la opción.</param>
    Friend Sub New(ByVal Codigo As Integer)
        Me.Codigo = Codigo
        Descripcion = "Confirmar Venta Actual"
    End Sub

End Class
